Lib/site.py

```diff
@@ -74,6 +74,11 @@
 import builtins
 import _sitebuiltins
 
+
+if os.environ.get("PYTHONSITEDEBUG", ""):
```

Comment:
Any PYTHON env var must be ignored when ignore environment is set.

```suggestion
if not sys.flags.ignore_environment and os.environ.get("PYTHONSITEDEBUG"):
```